File: MvvmListview/src/main/java/com/ibot/cyranosystems/mvvmrecycleview/MainActivity.java
package com.ibot.cyranosystems.mvvmrecycleview;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.ibot.cyranosystems.mvvmrecycleview.adapter.NewsAdapter;
import com.ibot.cyranosystems.mvvmrecycleview.databinding.ActivityMainBinding;
import com.ibot.cyranosystems.mvvmrecycleview.viewmodel.NewsModel;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private NewsModel newsModel;
    private ActivityMainBinding activityMainBinding;
    private ArrayList<NewsModel> newsModelArrayList;
    private NewsAdapter newsAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        activityMainBinding=DataBindingUtil.setContentView(this,R.layout.activity_main);
        newsModel=new NewsModel();
        newsModelArrayList=newsModel.getArrayListNews();
        newsAdapter=new NewsAdapter(this,newsModelArrayList);
        activityMainBinding.listview.setAdapter(newsAdapter);
    }
}

File: BottomNavigationKotlin/src/main/java/com/ibot/cyranosystems/bottomnavigationkotlin/MainActivity.kt
package com.ibot.cyranosystems.bottomnavigationkotlin

import android.os.Bundle
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView

class MainActivity : AppCompatActivity() {

private var content: FrameLayout? = null

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_home -> {

                val fragment = FragmentOne.Companion.newInstance()
                addFragment(fragment)

                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_dasboard -> {
                val fragment = FragmentTwo()
                addFragment(fragment)
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_notification -> {
                val fragment = FragmentThree()
                addFragment(fragment)
                return@OnNavigationItemSelectedListener true
            }
        }
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        content = findViewById(R.id.content)
        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)

        val fragment = FragmentOne.Companion.newInstance()
        addFragment(fragment)
    }
    /**
     * add/replace fragment in container [FrameLayout]
     */
    private fun addFragment(fragment: Fragment) {
        supportFragmentManager
                .beginTransaction()
                .setCustomAnimations(R.anim.design_bottom_sheet_slide_in, R.anim.design_bottom_sheet_slide_out)
                .replace(R.id.content, fragment, fragment.javaClass.simpleName)
                .commit()
    }
}
